### PR TITLE
dynamic route for grpc proxy

### PR DESCRIPTION
**Pull request by serprex**

### CODE REVIEW

**Review by heavycrystal — APPROVED** [review]

File: ui/app/alert-config/new.tsx [hunk 1]
```diff
@@ -197,7 +197,7 @@ export function NewConfig(alertProps: AlertConfigProps) {
     );
 
     setLoading(true);
-    const createRes = await fetch('/api/alert-config', {
+    const createRes = await fetch('/api/alerts/config', {
```
Comment (heavycrystal) [inline ui/app/alert-config/new.tsx:200]: `v1`?

File: ui/app/scripts/page.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import ScriptsTable from './list';
 
 export default function ScriptsPage() {
   const { data: res, isLoading } = useSWR<GetScriptsResponse>(
-    '/api/scripts?id=-1',
+    '/api/v1/scripts/-1',
```
Comment (heavycrystal) [inline ui/app/scripts/page.tsx:12]: now this case looks kind of weird

File: ui/app/api/v1/[...slug]/route.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+export async function DELETE(
+  request: NextRequest,
+  { params }: { params: { slug: string[] } }
+) {
+  const flowServiceAddr = GetFlowHttpAddressFromEnv();
+  let url = `${flowServiceAddr}/v1/${params.slug
+    .map((x) => encodeURIComponent(x))
+    .join('/')}`;
+  if (request.nextUrl.searchParams?.size) {
```
Comment (heavycrystal) [inline ui/app/api/v1/[...slug]/route.ts:46]: could embed this in url directly with coalesce to ''

File: ui/app/alert-config/new.tsx (hunk 1 above)
Comment (serprex) [reply ui/app/alert-config/new.tsx]: right

File: ui/app/api/v1/[...slug]/route.ts (hunk 3 above)
Comment (serprex) [reply ui/app/api/v1/[...slug]/route.ts:46]: trouble is I'm trying to avoid encoding empty searchParams to `?`

**Comment by Amogh-Bharadwaj** [comment]
undergoing functional testing

File: ui/app/mirrors/create/handlers.ts [hunk 4]
```diff
@@ -192,7 +192,7 @@ export const handleCreateCDC = async (
   }
 
   setLoading(true);
-  const res = await fetch('/api/mirrors/cdc', {
+  const res = await fetch('/api/v1/mirrors/cdc/create', {
```
Comment (Amogh-Bharadwaj) [inline ui/app/mirrors/create/handlers.ts]:
```suggestion
  const res = await fetch('/api/v1/flows/cdc/create', {
```

File: ui/app/mirrors/create/handlers.ts [hunk 5]
```diff
@@ -288,7 +288,7 @@ export const handleCreateQRep = async (
   }
 
   setLoading(true);
-  const res = await fetch('/api/mirrors/qrep', {
+  const res = await fetch('/api/v1/mirrors/qrep/create', {
```
Comment (Amogh-Bharadwaj) [inline ui/app/mirrors/create/handlers.ts]:
```suggestion
  const res = await fetch('/api/v1/flows/qrep/create', {
```

**Review by Amogh-Bharadwaj — APPROVED** [review]

**Pull request merged**